ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -177,6 +186,35 @@ S3Connection::ObjectExists(const std::string &objectKey, bool &exists, size_t &o
      * Fix later, return a proper error code. Not sure if we always have
      * outcome.GetError().GetResponseCode()
      */
-    std::cerr << "Error in ObjectExists." << std::endl;
     return (1);
 }
+
+/*
+ * BucketExists --
+ *     Checks whether the bucket configured for the class is accessible to us or not.
+ */
+int
+S3Connection::BucketExists(bool &exists) const
+{
+    exists = false;
+
+    Aws::S3Crt::Model::HeadBucketRequest request;
+    request.WithBucket(_bucketName);
```

Comment:
Given its usage and if you really on a `_member`, let's make it private. Otherwise you can leave it public but add `const std::string &bucketName` as an arg of the function.

Reply:
Moved to private.